up_client_socket/rust/utransport-socket/src/lib.rs

```diff
@@ -0,0 +1,479 @@
+        )?;
+
+        // Check the type of UAttributes (Publish / Request / Response)
+        match attributes
+            .type_
+            .enum_value()
+            .map_err(|_| UStatus::fail_with_code(UCode::INTERNAL, "Unable to parse type"))?
+        {
+            UMessageType::UMESSAGE_TYPE_PUBLISH => {
+                println!("UMESSAGE_TYPE_PUBLISH sending data");
+                UAttributesValidators::Publish
+                    .validator()
+                    .validate(&attributes)
+                    .map_err(|e| {
+                        UStatus::fail_with_code(
+                            UCode::INVALID_ARGUMENT,
+                            format!("Wrong Publish UAttributes {e:?}"),
+                        )
+                    })?;
+
+                match socket_clone.write_all(&umsg_serialized) {
+                    Ok(()) => Ok(()),
+                    Err(err) => Err(UStatus::fail_with_code(
+                        UCode::UNAVAILABLE,
+                        format!("Dispatcher communication issue: {err:?}"),
+                    )),
+                }
+            }
+            UMessageType::UMESSAGE_TYPE_REQUEST => {
+                UAttributesValidators::Request
+                    .validator()
+                    .validate(&attributes)
+                    .map_err(|e| {
+                        UStatus::fail_with_code(
+                            UCode::INVALID_ARGUMENT,
+                            format!("Wrong Request UAttributes {e:?}"),
+                        )
+                    })?;
+
+                match socket_clone.write_all(&umsg_serialized) {
+                    Ok(()) => Ok(()),
+                    Err(err) => Err(UStatus::fail_with_code(
+                        UCode::UNAVAILABLE,
+                        format!("Dispatcher communication issue: {err:?}"),
+                    )),
+                }
+            }
+            UMessageType::UMESSAGE_TYPE_RESPONSE => {
+                UAttributesValidators::Response
+                    .validator()
+                    .validate(&attributes)
+                    .map_err(|e| {
+                        UStatus::fail_with_code(
+                            UCode::INVALID_ARGUMENT,
+                            format!("Wrong Response UAttributes {e:?}"),
+                        )
+                    })?;
+                match socket_clone.write_all(&umsg_serialized) {
+                    Ok(()) => Ok(()),
+                    Err(err) => Err(UStatus::fail_with_code(
+                        UCode::UNAVAILABLE,
+                        format!("Dispatcher communication issue: {err:?}"),
+                    )),
+                }
+            }
+            UMessageType::UMESSAGE_TYPE_UNSPECIFIED | UMessageType::UMESSAGE_TYPE_NOTIFICATION => {
+                Err(UStatus::fail_with_code(
+                    UCode::INVALID_ARGUMENT,
+                    "Wrong Message type in UAttributes",
```

Comment:
Notification is not really wrong, right? Maybe separate these cases to distinguish them from "not implemented" vs "invalid type"?

Reply:
right, changed it to an INTERNAL code with "not implemented"